modules/apps/commerce/commerce-product-test/src/testFunctional/tests/GuestCheckout.testcase

```diff
@@ -695,6 +695,150 @@ definition {
+	@priority = 5
+	test CannotGuestAddressBeInheritedForLoggedUser {
+		property portal.acceptance = "true";
+
+		task ("Given a Minium site with a buyer user is created") {
+			CommerceAccelerators.initializeNewSiteViaAccelerator(siteName = "Minium");
+
+			CommerceEntry.addAccountEntryUser(
+				accountName = "Commerce Account",
+				accountType = "Business",
+				agreeToTermsAndAnswerReminderQuery = "true",
+				createNewAccount = "true",
+				requireReset = "false",
+				userEmailAddress = "buyer@liferay.com",
+				userFirstName = "Buyer",
+				userLastName = "User",
+				userRole = "Buyer",
+				userScreenName = "buyeruser",
+				userSiteMembership = "Minium");
+		}
+
+		task ("And an account address is added") {
+			Account.openAccountsAdmin();
+
+			LexiconEntry.gotoEntry(rowEntry = "Commerce Account");
+
+			Account.gotoAssigneeTab(assigneeTab = "Addresses");
+
+			Account.addAddress(
+				addressCity = "Test City",
+				addressCountry = "United States",
+				addressName = "Test Address",
+				addressPostalCode = "Test Zip Code",
+				addressRegion = "California",
+				addressStreet1 = "Test Street");
+		}
+
+		task ("And 'Guest Checkout' is enabled for the site") {
+			CommerceChannels.openChannelsAdmin();
+
+			CommerceNavigator.gotoEntry(entryName = "Minium Portal");
+
+			Check.checkToggleSwitch(
+				key_toggleSwitchLabel = "Guest Checkout",
+				locator1 = "ToggleSwitch#ANY_TOGGLE_SWITCH");
+
+			Button.clickSave();
+		}
+
+		task ("And the Manage Addresses permission of Accounts is removed for the Buyer role") {
+			Permissions.removePermissionViaAPI(
+				groupName = "Minium",
+				resourceAction = "MANAGE_ADDRESSES",
+				resourceName = "com.liferay.account.model.AccountEntry",
+				roleTitle = "Buyer",
+				roleType = "account");
+		}
+
+		task ("And the View permission of Catalog, Pending Orders, Checkout pages for Guest are checked") {
+			for (var pageName : list "Catalog,Pending Orders,Checkout") {
+				PagesAdmin.openPagesAdmin(siteURLKey = "minium");
+
+				if (${pageName} == "Checkout") {
+					PagesAdmin.gotoPageEllipsisMenuItem(
+						menuItem = "Permissions",
+						pageName = ${pageName},
+						parentPagesHierarchy = "Pending Orders");
+				}
+				else {
+					PagesAdmin.gotoPageEllipsisMenuItem(
+						menuItem = "Permissions",
+						pageName = ${pageName});
+				}
+
+				Role.definePermissionPG(
+					permissionDefinitionKey = "CONFIGURATION_PERMISSIONS_VIEW_CHECKBOX",
+					roleTitle = "guest");
+			}
+		}
+
+		task ("And the guest user adds a product to the cart") {
+			User.logoutPG();
+
+			Navigator.openWithAppendToBaseURL(urlAppend = "web/minium/catalog");
+
+			FrontStore.addInitializerProductToCart(
+				productName = "U-Joint",
+				productQuantity = 1);
+		}
+
+		task ("And adds an address in the checkout page and moves to next step") {
+			CommerceCheckout.gotoCheckoutViaMiniCart();
+
```

Comment:
Based on the ticket's description, we can see that the list of available addresses is either disabled/not showing any address. Could we add a check for this at this point? Thanks!

Reply:
Hi @lorenzolapescara, the address list selector doesn't show for the guest user, so I just added an assertion for it as the selector is not present, thanks!
![result](https://github.com/liferay-commerce/liferay-portal/assets/85482945/33cf0693-bf3c-4c2b-bf5d-ad7bb99654a6)